connection.go

```diff
@@ -756,3 +756,19 @@ func (sc *snowflakeConn) FetchResult(ctx context.Context, qid string) (driver.Ro
 type ResultFetcher interface {
 	FetchResult(ctx context.Context, qid string) (driver.Rows, error)
 }
+
+func (sc *snowflakeConn) FetchMonitoringResult() (
+	*monitoringResult, error) {
+	if sc.rest == nil {
+		return nil, driver.ErrBadConn
+	}
+
+	// set the fake runtime just to bypass fast query
+	monitoringResult := mkMonitoringFetcher(sc, sc.QueryID, time.Minute*10)
```

Comment:
Are there any concurrency issues with using this? I remember the first time @mhseiden added the shim to fetch monitoring data we needed up with concurrency issue later one (@agam looked into it if I remember correctly).

Reply:
This is the fix for a data-race I had made earlier: https://github.com/sigmacomputing/gosnowflake/pull/32/commits/0540a663df0afd11f02fdc0a122e3ac9513a53f7

The race was between the monitoring-fetcher and the _connection_. It's independent of whether or not we have a `rows` object, so we should be fine here.